### PR TITLE
dev/core#1295 Add is_active field to civicrm_status_preference

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Add is_active field to civicrm_status_pref per discussions in NYC this is to be api-accessible but not UI exposed. 

Importantly the is_active field is checked BEFORE attempting the check rather than after

Before
----------------------------------------
No way to stop a check from running at all

After
----------------------------------------
is_active field can be set to zero to disable

Technical Details
----------------------------------------
For devs to disable a status check they would need to add or edit a row in civicrm_status_check for that  check - e.g 
```
 $this->callAPISuccess('StatusPreference', 'create', [
      'name' => 'checkDefaultMailbox',
      'is_active' => 0,
    ]);
```

Comments
----------------------------------------

### CODE REVIEW

**Comment by seamuslee001** [comment]
@eileenmcnaughton Jenkins is displeased https://test.civicrm.org/job/CiviCRM-Core-PR/29164/checkstyleResult/new/

File: CRM/Utils/Check/Component.php [hunk 1]
```diff
@@ -25,14 +25,44 @@
  +--------------------------------------------------------------------+
  */
 
-
+use Civi\Api4\StatusPreference;
```
Comment (seamuslee001) [inline CRM/Utils/Check/Component.php]: @eileenmcnaughton why is this needed?

File: CRM/Utils/Check/Component.php [hunk 2]
```diff
@@ -25,14 +25,44 @@
...
+   * @return array
+   *
+   * @throws \API_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
+   */
+  public function getChecksConfig() {
+    if (empty($this->checksConfig)) {
+      $this->checksConfig = Civi::cache('checks')->get('checksConfig', []);
+      if (empty($this->checksConfig)) {
+        $this->checksConfig = StatusPreference::get()->execute()->indexBy('name');
```
Comment (seamuslee001) [inline CRM/Utils/Check/Component.php]: @eileenmcnaughton wouldn't you route through civicrm_api4 or similar or something?

File: CRM/Utils/Check/Component.php [hunk 3]
```diff
@@ -47,24 +78,49 @@ public function isEnabled() {
...
    *   [CRM_Utils_Check_Message]
+   *
+   * @throws \API_Exception
+   * @throws \Civi\API\Exception\UnauthorizedException
    */
   public function checkAll() {
     $messages = [];
     foreach (get_class_methods($this) as $method) {
-      if ($method !== 'checkAll' && strpos($method, 'check') === 0) {
+      $r = $this->isDisabled($method);
```
Comment (seamuslee001) [inline CRM/Utils/Check/Component.php]: @eileenmcnaughton what is this used for at all?

Comment (eileenmcnaughton) [reply CRM/Utils/Check/Component.php]: @seamuslee001 thanks - fixed

**Comment by seamuslee001** [comment]
I've added merge ready as i think this is right but we may want to have a discussion about whether we want to be standardising on the OOP style API4 coding or use the traditional `civicrm_api4` style. @colemanw any thoughts?
